frontera/contrib/backends/remote/messagebus.py

```diff
@@ -93,19 +97,65 @@ def states(self):
         return None
 
 
+class BufferedMixin(object):
+    def __init__(self, *args, **kwargs):
+        super_self = super(BufferedMixin, self)
+
+        super_self.__init__(*args, **kwargs)
+
+        self._buffer = OverusedBuffer(
+            super_self.get_next_requests,
+            self._logger.debug,
+        )
+
+    def get_next_requests(self, max_n_requests, **kwargs):
+        return self._buffer.get_next_requests(max_n_requests, **kwargs)
+
+
+class AggregatePerHostMixin(object):
+    def add_seeds(self, seeds):
+        per_host = aggregate_per_host(seeds)
+
+        # TODO:2017-02-19: implement support of None key in partitioners
+        per_host.pop(None, None)
+
+        for key, host_links in six.iteritems(per_host):
+            message = self._encoder.encode_add_seeds(host_links)
+
+            self.spider_log_producer.send(key, message)
+
+    def links_extracted(self, request, links):
+        per_host = aggregate_per_host(links)
+
+        # TODO:2017-02-19: implement support of None key in partitioners
+        per_host.pop(None, None)
+
+        for key, host_links in six.iteritems(per_host):
+            message = self._encoder.encode_links_extracted(request, host_links)
+
+            self.spider_log_producer.send(key, message)
+
+
+class MessageBusBackend(
+    BufferedMixin,
+    AggregatePerHostMixin,
+    GenericMessageBusBackend,
+):
+    pass
+
+
 def aggregate_per_host(requests):
-    per_host = dict()
-    for link in requests:
-        if b'fingerprint' not in link.meta[b'domain']:
-            continue
-        host_fprint = link.meta[b'domain'][b'fingerprint']
-        if host_fprint not in per_host:
-            per_host[host_fprint] = []
-        per_host[host_fprint].append(link)
+    per_host = defaultdict(list)
+
+    for request in requests:
+        key = get_host_fprint(request)
```

Comment:
The above code skips the link, if fingerprint wasn't found in meta[b'domain'], but your implementation adds it with None key, which will cause MessageBus to bark.